### PR TITLE
Implemented functionality for ChronicleSession

### PR DESCRIPTION
**Pull request by young-k**
Chronicle is our Register replacement, this sends logs once there's 10 of a given event type and uploads it to S3.

I've tested pretty extensively by sending mock logs, but will test more tomorrow in prep for a demo.

### CODE REVIEW

File: src/ChronicleSession.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    cacheSize: number = 10,
+  ) {
+    this.app = app;
+    this.logMap = new Map();
+    this.cacheSize = cacheSize;
+
+    config.update({
+      accessKeyId: accessKey,
+      secretAccessKey: secretKey,
+    })
```
Comment (kq-li) [inline src/ChronicleSession.js]: missing semicolon

File: src/ChronicleSession.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    this.s3 = new S3();
+  }
+
+  async writeLogs(eventName: string, eventType: ParquetSchema) {
+    const filename = `${Date.now()}.parquet`;
+    const schema = new ParquetSchema(eventType);
+    const writer = await ParquetWriter.openFile(schema, `${PATH}/${filename}`);
+
+    const logs = this.logMap.get(eventName);
+    if (logs === undefined) { // sanity check
```
Comment (kq-li) [inline src/ChronicleSession.js]: might want `==` here to check for null too

File: src/ChronicleSession.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+      return;
+    }
+
+    logs.forEach((log) => { writer.appendRow(log) });
```
Comment (kq-li) [inline src/ChronicleSession.js]: `log => writer.appendRow(log)`?

File: src/ChronicleSession.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    writer.close();
+    this.logMap.set(eventName, []);
+
+    const params = {
+      Bucket: BUCKET,
+      Body : createReadStream(`${PATH}/${filename}`),
+      Key : `${this.app}/${eventName}/${filename}`
```
Comment (kq-li) [inline src/ChronicleSession.js]: no spaces before colons

File: src/ChronicleSession.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    };
+    await this.s3.upload(params).promise();
+  }
+
+  async log(eventName: string, eventType: ParquetSchema, event: Object) {
+    const logs = this.logMap.get(eventName);
+    if (logs === undefined) {
```
Comment (kq-li) [inline src/ChronicleSession.js]: `==`?

File: src/index.js [hunk 6]
```diff
@@ -1,5 +1,6 @@
 // @flow
-export { default as AppDevRouter} from './AppDevRouter';
 export { default as AppDevAPI }  from './AppDevAPI';
-export { default as RegisterSession } from './RegisterSession';
+export { default as AppDevRouter} from './AppDevRouter';
```
Comment (kq-li) [inline src/index.js]: consistency: either spaces or no spaces between braces (i prefer no spaces personally, we should write an appdev style guide bc airbnb has some questionable recommendations)

**Review by kq-li — APPROVED** [review]
LGTM, just some style gotchas

File: src/ChronicleSession.js (hunk 3 above)
Comment (ilikesymmetry) [reply src/ChronicleSession.js]: writer.appendRow(log) is spec from the ParquetWriter to add a row in the parquet file

File: src/index.js (hunk 6 above)
Comment (young-k) [reply src/index.js]: nah we should have spaces

File: src/ChronicleSession.js (hunk 3 above)
Comment (kq-li) [reply src/ChronicleSession.js]: :+1: , i just prefer the shorthand for the one-argument, one-line arrow function

File: src/index.js (hunk 6 above)
Comment (kq-li) [reply src/index.js]: in that case `AppDevRouter }`

**Comment by ilikesymmetry** [comment]
Don't forget to update the main README on appdev.js

**Pull request merged**